### DISCUSSION TITLE
建议提供一些文档让开发者来写一些开发MAA pipeline中的心得

### DISCUSSION THREAD

**Discussion by BrokenArrow1404** [MaaXYZ/MaaFramework#506] [Ideas]
建议提供一些文档让开发者来写一些开发MAA pipeline中的心得， 类似开发实践总结之类的。（似乎1999的repo里有一些，但是感觉不够完整）

**Comment by Windsland52** [comment]
个人感觉还是读明白pipeline协议文档后在自行实践中形成自己的风格较好

**Comment by Windsland52** [comment]
由于很容易出现很多奇怪细节导致的问题，个人认为找bug的工具/能力/经验更重要

**Comment by BrokenArrow1404** [comment]
> 由于很容易出现很多奇怪细节导致的问题，个人认为找bug的工具/能力/经验更重要

是的，但是MAA的其中一部分用户似乎是没有太多编程经验的小白，所以我感觉从小白的角度出发可能会有这个需求

不过也许小白通过第一次参加open source 多敲敲代码就进化了（

**Comment by Windsland52** [comment]
编写pipeline的话我感觉pipeline协议文档（就像字典）已经足够简洁而又详尽了，再写心得啥的也是对每个条目进行扩写之类的。

**Comment by Windsland52** [comment]
突然想起来，我在[个人博客](https://windsland52.github.io/archives/)里倒是写过几篇开发日志（不过可能有的想法不对），仅供参考。

**Comment by neko-para** [comment]
理论上, 直接使用对应项目的discussion就挺不错的(
需要的话完全可以在主页上给链接嘛

**Comment by BrokenArrow1404** [comment]
> 突然想起来，我在[个人博客](https://windsland52.github.io/archives/)里倒是写过几篇开发日志（不过可能有的想法不对），仅供参考。

啊是的，我感觉你的这些记录就是我说的那种
举个最简单的例子：应该让每一个task list做到模块化（也就是从游戏主页开始，最终回到游戏主页，这样可以独立运行），这种设计思路可能不是一个完全没有coding经验的人一下子就能想到的
不恰当的类比也许就是design pattern？

**Comment by Windsland52** [comment]
这种可能需要再沉淀沉淀才能下笔写
